### PR TITLE
389: add kima modal

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Adding a modal to be shown based on bridge-status.

- [x] - Merged the learn pressable's into one component: LearnButton
- [x] - Add modal
- [x] - add story

About # (link your issue here)
https://github.com/GoodDollar/GoodProtocolUI/issues/389

# How Has This Been Tested?

### CODE REVIEW

**Comment by sirpy** [comment]
@L03TJ3 i'm not sure you are reseting the modal state. ie would it open again on a second tx?

**Comment by L03TJ3** [comment]
image
Current design
@patpedrosa kindly confirm before I push this

File: packages/good-design/src/core/buttons/LearnButton.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { Image } from "../images";
+import BackToSchool from "../../assets/images/backtoschool.png";
+
+type learnLinks = {
+  transactions: string;
+  bridging: string;
+  network: string;
+  signing: string;
+};
+const links: learnLinks = {
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/buttons/LearnButton.tsx:14]: why not compose onto single map with { link: string; label: string } value ?

e.g. 

```typescript
const links: Record<string, { link: string; label: string }> = {
  transactions: {
    label: "How do transactions work?",
    link: "https://www.notion.so/gooddollar/How-do-transactions-work-in-web3-ccf11b4e16874a1682722a20c4e24742"
  },
  // ...
}
```

File: packages/good-design/src/core/web3/KimaModal.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+interface KimaModalProps {
+  success: boolean | undefined;
+  networks: BridgeNetworks;
+  resetState: any;
+  children: any;
+}
+
+const KimaModalCopy = (success: boolean, networks: BridgeNetworks) => ({
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/web3/KimaModal.tsx:18]: define as 2 components: KimaModalHeader / KimaModalBody receiving such props

File: packages/good-design/src/core/web3/KimaModal.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const { Modal, showModal } = useModal();
+  const kimaCopy = KimaModalCopy(success ?? false, networks);
+
+  useEffect(() => {
+    if (success !== undefined) {
+      showModal();
+    }
+  }, [success, showModal]);
+
+  const onClose = useCallback(() => {
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/web3/KimaModal.tsx:41]: redundant, just pass resetState as onClose cb

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/core/buttons/LearnButton.tsx (hunk 1 above)
Comment (L03TJ3) [reply packages/good-design/src/core/buttons/LearnButton.tsx]: done

File: packages/good-design/src/core/web3/KimaModal.tsx (hunk 2 above)
Comment (L03TJ3) [reply packages/good-design/src/core/web3/KimaModal.tsx]: 1 component, KimaModalHeader

Chose to do learn button directly as I didn't see the need for it to be a separate component if there are no props and/or deps

File: packages/good-design/src/core/web3/KimaModal.tsx (hunk 3 above)
Comment (L03TJ3) [reply packages/good-design/src/core/web3/KimaModal.tsx]: ye, quite rendunant. :) 
fixed

**Comment by L03TJ3** [comment]
Final view: 
image

File: packages/good-design/src/core/web3/KimaModal.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  useEffect(() => {
+    if (success !== undefined) {
+      showModal();
+    }
+  }, [success, showModal]);
+
+  return (
+    <React.Fragment>
+      <Modal header={header} body={<LearnButton source="bridging" />} onClose={resetState} closeText="x" />
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/web3/KimaModal.tsx:46]:
```suggestion
      <Modal header={<KimaModalHeader success={success ?? false} networks={networks} />} body={<LearnButton source="bridging" />} onClose={resetState} closeText="x" />
```

File: packages/good-design/src/core/web3/KimaModal.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        {success ? `Success!` : `Oops!`}
+      </Title>
+      <Text color="goodGrey.500" fontSize="sm" fontFamily="subheading">
+        {success
+          ? `You have succesfully bridged from ${networks.origin} to ${networks.destination}`
+          : `Something went wrong while trying to bridge your tokens`}
+      </Text>
+    </Box>
+  )
+});
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply packages/good-design/src/core/web3/KimaModal.tsx:32]: this is how exactly I've asked to implement

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**